src/react/components/StyleForm.jsx

```diff
@@ -0,0 +1,348 @@
+                    <Form.Item name='fill' label='Pisteen täyttöväri' { ...formLayout }>
+                        <ColorPicker
+                            onChange={ (event) => this.updateState('fill.color', event.target.value) }
+                            defaultValue={ this.state.fill.color }
+                        />
+                    </Form.Item>
+                    : false
+                }
+            </Row>
+        );
+    }
+
+    _getSizeControl () {
+        return (
+            <Row>
+                <Form.Item name='size' label='Size' initialValue={ 3 } { ...formLayout }>
+                    <InputNumber
+                        min={ 1 }
+                        max={ 5 }
+                        formatter={ sizeFormatter }
+                        parser={ sizeFormatter }
+                        onChange={ (value) => {
+                            this.updateState('stroke.width', value);
+                            this.updateState('stroke.area.width', value);
+                            this.updateState('image.size', value);
+                        } } />
+                </Form.Item>
+            </Row>
+        );
+    }
+
+    _buildStyleList (options) {
+        return (
+            <Form.Item label='Tyylit' { ...formLayout } name={ 'styleListSelector' }>
+                <Select onChange={ this._populateWithStyle }>
+                    { options.map((singleOption) => {
+                        return (
+                            <Select.Option
+                                value={ singleOption.value}
+                                key={ singleOption.value }
+                            >
+                                { singleOption.value }
+                            </Select.Option>
+                        );
+                    }) }
+                </Select>
+            </Form.Item>
+        );
+    }
+
+    render () {
+        return (
+            <StaticForm ref={ this.ref }>
+                <Space direction='vertical'>
+                    <Card>
+                        { this._buildStyleList( this.props.styleList ) }
+
+                        <Form.Item label='Muoto' { ...formLayout } name={ 'format' } initialValue={ this.state.format }>
+                            <TabSelector { ...formLayout } onChange={ this.changeTab } key={ 'formatSelector' } name='format' >
+                                <Radio.Button value='point'>Piste</Radio.Button>
+                                <Radio.Button value='line'>Viiva</Radio.Button>
+                                <Radio.Button value='area'>Alue</Radio.Button>
+                            </TabSelector>
+                        </Form.Item>
+
+                        <Card>
+                            { this._getColorPickers() }
+
+                            { this._getCurrentTab( this.state.format ) }
+
+                            { this._getSizeControl() }
+
+                            { this._createPreview() }
```

Comment:
Color pickers, size controls and previews should all be in the "tab" components. The current tab can then be shown somewhat like in the layer adding wizard instead of calling a method. It's easier to see what's inside the `<Card>` that way: https://github.com/oskariorg/oskari-frontend/blob/master/bundles/admin/admin-layereditor/view/LayerWizard/LayerWizard.jsx